src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeCookieAction.cs

```diff
@@ -2,21 +2,80 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Rewrite.Internal.UrlActions
 {
     public class ChangeCookieAction : UrlAction
     {
-        public ChangeCookieAction(string cookie)
+        private readonly Func<DateTimeOffset> _timeSource;
+        private CookieOptions _cachedOptions;
+
+        public ChangeCookieAction(string name)
+            : this(name, () => DateTimeOffset.UtcNow)
         {
-            // TODO
-            throw new NotImplementedException("Changing the cookie is not implemented");
         }
 
+        // for testing
+        internal ChangeCookieAction(string name, Func<DateTimeOffset> timeSource)
+        {
+            _timeSource = timeSource;
+
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(nameof(name));
+            }
+
+            Name = name;
+        }
+
+        public string Name { get; }
+        public string Value { get; set; }
+
+        public string Domain { get; set; }
+
+        /// <summary>
```

Comment:
It's internal API, so we generally don't work about XML doc. I've only added it here because it's important to capture that this is **minutes**